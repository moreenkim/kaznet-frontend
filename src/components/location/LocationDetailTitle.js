// Renders the detail page title section
import React, { Component } from "react";
import FontAwesomeIcon from "@fortawesome/react-fontawesome";
import {
  Row,
  Col,
  Badge,
  Modal,
  ModalHeader,
  ModalFooter,
  Button
} from "reactstrap";
import { Link } from "react-router-dom";
import Moment from "react-moment";

import "../page/DetailTitle.css";

export default class LocationDetailTitle extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({
      modal: !this.state.modal
    });
  }

  render() {
    return (
      <section className="detail-page-title">
        <Col sm="12" md={{ size: 8, offset: 2 }}>
          <div>
            <h1 className="kaznet-detail-title">
              <Link to="/locations" className="kaznet-header-link">
                Locations
              </Link>{" "}
              > {this.props.location.attributes.name}
            </h1>
            <p className="kaznet-creation-detail">
              {"By "}
              {this.props.location.attributes.created_by_name},{" "}
              <Moment format="DD-MM-YYYY">
                {this.props.location.attributes.created}
              </Moment>
            </p>
            <Col md="12">
              <Row className="kaznet-action-links">
                <Link
                  to={`/locations/${this.props.location.id}/edit`}
                  className="action-link"
                >
                  EDIT
                </Link>&nbsp;&nbsp;|&nbsp;&nbsp;
                <Button
                  color="link"
                  className="remove_button_css action-link action-link-alert"
                  onClick={this.toggle}
                >
                  DELETE LOCATION
                </Button>
                <Modal
                  isOpen={this.state.modal}
                  toggle={this.toggle}
                  className={this.props.className}
                >
                  <ModalHeader toggle={this.toggle}>
                    Are you sure you want to delete this Location?
                  </ModalHeader>
                  <ModalFooter>
                    <Link
                      to={`/locations/${this.props.location.id}/delete`}
                      className="btn btn-danger"
                      onClick={this.toggle}
                    >
                      Delete Location
                    </Link>
                    <Button color="secondary" onClick={this.toggle}>
                      Cancel
                    </Button>
                  </ModalFooter>
                </Modal>
              </Row>
            </Col>
          </div>
        </Col>
      </section>
    );
  }
}
