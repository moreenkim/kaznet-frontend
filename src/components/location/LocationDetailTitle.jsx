// Renders the detail page title section
import React, { Component } from 'react';
import { Row, Col, Modal, ModalHeader, ModalFooter, Button } from 'reactstrap';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import '../page/DetailTitle.scss';

export default class LocationDetailTitle extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    const { modal } = this.state;

    this.setState({
      modal: !modal
    });
  }

  render() {
    const { location, className } = this.props;
    const { modal } = this.state;

    const locationData = location.attributes;
    const locationID = location.id;

    return (
      <section className="detail-page-title">
        <Col sm="12" md={{ size: 8, offset: 2 }}>
          <div>
            <h1 className="kaznet-detail-title">
              <Link to="/locations" className="kaznet-header-link">
                Locations
              </Link>
              &nbsp;&gt;&nbsp;
              {locationData.name}
            </h1>
            <Col md="12">
              <Row className="kaznet-action-links">
                <Link to={`/locations/${locationID}/edit`} className="action-link">
                  EDIT
                </Link>
                &nbsp;&nbsp;|&nbsp;&nbsp;
                {locationData.has_submissions === true && (
                  <div>
                    <p className="text-muted">DELETE LOCATION</p>
                  </div>
                )}
                {locationData.has_submissions === false && (
                  <div>
                    <Button
                      color="link"
                      className="remove_button_css action-link action-link-alert"
                      onClick={this.toggle}
                      aria-label="DELETE LOCATION"
                    >
                      DELETE LOCATION
                    </Button>
                    <Modal isOpen={modal} toggle={this.toggle} className={className}>
                      <ModalHeader toggle={this.toggle}>
                        Are you sure you want to delete this Location?
                      </ModalHeader>
                      <ModalFooter>
                        <Link
                          to={`/locations/${locationID}/delete`}
                          className="btn btn-danger"
                          onClick={this.toggle}
                        >
                          Delete Location
                        </Link>
                        <Button color="secondary" onClick={this.toggle} aria-label="Cancel">
                          Cancel
                        </Button>
                      </ModalFooter>
                    </Modal>
                  </div>
                )}
              </Row>
            </Col>
          </div>
        </Col>
      </section>
    );
  }
}

LocationDetailTitle.propTypes = {
  location: PropTypes.shape({
    attributes: {},
    id: PropTypes.number
  }).isRequired,
  className: PropTypes.string.isRequired
};
