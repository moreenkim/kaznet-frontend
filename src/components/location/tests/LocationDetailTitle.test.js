// Test LocationDetailTitle
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Router } from 'react-router';
import createBrowserHistory from 'history/createBrowserHistory';

import ErrorBoundary from '../../ErrorBoundary';
import { singleLocation } from '../../../store/locations/tests/fixtures';
import LocationDetailTitle from '../LocationDetailTitle';

const history = createBrowserHistory();

describe('components/location/LocationDetailTitle', () => {
  it('renders without crashing', () => {
    shallow(
      <ErrorBoundary>
        <LocationDetailTitle location={singleLocation} />
      </ErrorBoundary>
    );
  });
  it('renders the correct html', () => {
    const wrapper = mount(
      <ErrorBoundary>
        <Router history={history}>
          <LocationDetailTitle location={singleLocation} />
        </Router>
      </ErrorBoundary>
    );

    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.unmount();
  });
});
