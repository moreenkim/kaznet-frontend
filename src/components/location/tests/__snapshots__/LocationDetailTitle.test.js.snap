// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/location/LocationDetailTitle renders the correct html 1`] = `
<ErrorBoundary>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LocationDetailTitle
      location={
        Object {
          "attributes": Object {
            "country": "KE",
            "created": "2018-07-09T14:53:09.732230+03:00",
            "description": "Something",
            "geopoint": null,
            "location_type_name": null,
            "modified": "2018-07-09T14:53:09.732261+03:00",
            "name": "Isiolo Market X",
            "parent_name": null,
            "radius": null,
            "shapefile": null,
          },
          "id": "999",
          "relationships": Object {
            "location_type": Object {
              "data": null,
            },
            "parent": Object {
              "data": null,
            },
          },
          "type": "Location",
        }
      }
    >
      <section
        className="detail-page-title"
      >
        <Col
          md={
            Object {
              "offset": 2,
              "size": 8,
            }
          }
          sm="12"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <div
            className="col-sm-12 col-md-8 offset-md-2"
          >
            <div>
              <h1
                className="kaznet-detail-title"
              >
                <Link
                  className="kaznet-header-link"
                  replace={false}
                  to="/locations"
                >
                  <a
                    className="kaznet-header-link"
                    href="/locations"
                    onClick={[Function]}
                  >
                    Locations
                  </a>
                </Link>
                 
                &gt; 
                Isiolo Market X
              </h1>
              <p
                className="kaznet-creation-detail"
              >
                By 
                ,
                 
                <t
                  ago={false}
                  calendar={false}
                  decimal={false}
                  element="time"
                  filter={[Function]}
                  format="DD-MM-YYYY"
                  fromNow={false}
                  interval={60000}
                  onChange={[Function]}
                  toNow={false}
                  unit={null}
                  unix={false}
                  utc={false}
                >
                  <time
                    dateTime={"2018-07-09T11:53:09.732Z"}
                  >
                    09-07-2018
                  </time>
                </t>
              </p>
              <Col
                md="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-md-12"
                >
                  <Row
                    className="kaznet-action-links"
                    tag="div"
                  >
                    <div
                      className="kaznet-action-links row"
                    >
                      <Link
                        className="action-link"
                        replace={false}
                        to="/locations/999/edit"
                      >
                        <a
                          className="action-link"
                          href="/locations/999/edit"
                          onClick={[Function]}
                        >
                          EDIT
                        </a>
                      </Link>
                        |  
                      <Button
                        className="remove_button_css action-link action-link-alert"
                        color="link"
                        onClick={[Function]}
                        tag="button"
                      >
                        <button
                          className="remove_button_css action-link action-link-alert btn btn-link"
                          onClick={[Function]}
                          type="button"
                        >
                          DELETE LOCATION
                        </button>
                      </Button>
                      <Modal
                        autoFocus={true}
                        backdrop={true}
                        backdropTransition={
                          Object {
                            "mountOnEnter": true,
                            "timeout": 150,
                          }
                        }
                        centered={false}
                        fade={true}
                        isOpen={false}
                        keyboard={true}
                        modalTransition={
                          Object {
                            "timeout": 300,
                          }
                        }
                        onClosed={[Function]}
                        onOpened={[Function]}
                        role="dialog"
                        toggle={[Function]}
                        zIndex={1050}
                      />
                    </div>
                  </Row>
                </div>
              </Col>
            </div>
          </div>
        </Col>
      </section>
    </LocationDetailTitle>
  </Router>
</ErrorBoundary>
`;
